cfme/common/provider.py

```diff
@@ -222,30 +224,41 @@ def validate(self, db=True):
         client = self.get_mgmt_system()
 
         # If we're not using db, make sure we are on the provider detail page
-        if not db:
+        if ui:
             self.load_details()
 
         # Initial bullet check
-        if self._do_stats_match(client, self.STATS_TO_MATCH, db=db):
+        if self._do_stats_match(client, self.STATS_TO_MATCH, ui=ui):
             client.disconnect()
             return
         else:
             # Set off a Refresh Relationships
-            self.load_details()
-            tb.select("Configuration", "Refresh Relationships and Power States", invokes_alert=True)
-            sel.handle_alert()
+            method = 'ui' if ui else None
+            self.refresh(method=method)
 
             refresh_timer = RefreshTimer(time_for_refresh=300)
             wait_for(self._do_stats_match,
                      [client, self.STATS_TO_MATCH, refresh_timer],
-                     {'db': db},
+                     {'ui': ui},
                      message="do_stats_match_db",
                      num_sec=1000,
                      delay=60)
 
         client.disconnect()
 
-    def _do_stats_match(self, client, stats_to_match=None, refresh_timer=None, db=True):
+    @variable(alias='rest')
+    def refresh(self):
+        rest_api = store.current_appliance.rest_api
+        col = rest_api.collections.providers.find_by(name=self.name)[0]
+        col.action.refresh()
```

Comment:
I suppose in order for this to have the intended effect, shouldn't the REST object be stored in the provider object itself?